jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/internal/HttpChannelState.java

```diff
@@ -437,7 +437,7 @@ public Runnable onFailure(Throwable x)
                 // Notify the failure listeners only once.
                 Consumer<Throwable> onFailure = _onFailure;
                 _onFailure = null;
-                Runnable invokeOnFailureListeners = onFailure == null ? null : () ->
+                Runnable invokeOnFailureListeners = onFailure == null || !getHttpConfiguration().isNotifyRemoteAsyncErrors() ? null : () ->
```

Comment:
I'm not sure there is a way to reliably discriminate errors. Maybe this feature is specific to ee* environments?

Reply:
In Jetty 11, the check for `notifyRemoteAsyncErrors` was done in the HTTP/2 and HTTP/3 layers, not generically.

I think we should restore that.

The idea would be to introduce an `onRemoteFailure()` method that is only called from H2 and H3, and have this new method delegate to `onFailure()` with an extra parameter, so that pending reads and writes are notified, but listeners are only invoked if the extra parameter is true.

Reply:
Pushed an attempt at that; two observations:

 - having that notification filtering logic in core means it's not needed in ee* anymore
 - H3 does not have client resets?!